File: benchmarks/fibonacci/python/fib.py
import sys


def fib(n):
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)


def main():
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 32
    assert fib(10) == 55, "self-check failed: fib(10) != 55"
    print(fib(n))


if __name__ == "__main__":
    main()

File: benchmarks/fibonacci/c/fib.c
#include <stdio.h>
#include <stdlib.h>

long fib(int n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

int main(int argc, char **argv) {
    int n = argc > 1 ? atoi(argv[1]) : 32;
    if (fib(10) != 55) {
        fprintf(stderr, "self-check failed: fib(10) != 55\n");
        return 1;
    }
    printf("%ld\n", fib(n));
    return 0;
}
